src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -742,28 +756,55 @@ private void transferAlignmentInfoToPairedRead(final SAMRecord firstUnaligned, f
             transferAlignmentInfoToFragment(secondUnaligned, secondAligned, isContaminant, needsSafeReverseComplement);
         }
         if (isClipOverlappingReads()) {
-            clipForOverlappingReads(firstUnaligned, secondUnaligned);
+            clipForOverlappingReads(firstUnaligned, secondUnaligned, clipOverlappingReadsOperator, read1Structure, read2Structure, adapterPairs);
         }
         SamPairUtil.setMateInfo(secondUnaligned, firstUnaligned, addMateCigar);
         if (!keepAlignerProperPairFlags) {
             SamPairUtil.setProperPairFlags(secondUnaligned, firstUnaligned, expectedOrientations);
         }
     }
 
+
+
     /**
-     * Checks to see whether the ends of the reads overlap and soft clips reads
-     * them if necessary.
+     * Checks to see whether the ends of the reads overlap and clips reads
+     * if necessary.
      */
-    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2) {
+    protected static void clipForOverlappingReads(final SAMRecord read1, final SAMRecord read2, final CigarOperator clippingOperator, final ReadStructure read1Structure, final ReadStructure read2Structure,
```

Comment:
the generalizability of having the clipping operator given as such seems like premature optimization...why not just a boolean hardClip ?